clang/test/3C/add_bounds.c

```diff
@@ -0,0 +1,25 @@
+// RUN: rm -rf %t*
+// RUN: 3c -base-dir=%S -alltypes -addcr %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: 3c -base-dir=%S -alltypes -addcr %s -- | %clang -c -fcheckedc-extension -x c -o /dev/null -
+// RUN: 3c -base-dir=%S -alltypes -output-dir=%t.checked %s --
+// RUN: 3c -base-dir=%t.checked -alltypes %t.checked/add_bounds.c -- | diff %t.checked/add_bounds.c -
+
+void foo(_Array_ptr<int> b, int n) { }
+// CHECK_ALL: void foo(_Array_ptr<int> b : count(n), int n) _Checked { }
```

Comment:
This is relying on bounds heuristics. Can you also make a version that generates bounds from a static array, and a malloc, and make sure these carry over as you expect, too?